### ISSUE TITLE
Nested routes and refresh problem

### ISSUE THREAD

**Issue by gfduszynski** [riot/route#113]
Hi! I am having trouble with nested routes. 
My issue looks very similar to the #86 

I'm using version 3.1.2.

When I refresh the page only the top most router handles the event,
nested tag route handlers are not called. 

Example:
```js
<app>
  <router>
    <route path="path/foo"><inner-tag /></route>
  </router>
</app>

<inner-tag>
  <p>{ name }</p>
  <script>
    this.name = 'loading';
    var r    = route.create()

    r('path/foo',    changeName   )

    function changeName(){
        self.update({
            name:  "Greg"
        })
    }
  </script>
</inner-tag>
```
When using links, history or pasting url I can only see 'loading'.
However if I manually remove 'o' and press enter, and then add 'o' 
and press enter 'changeName' gets called.

Calling inside 'inner-tag'
```js
route.exec();
```
seems to solve the problem, but it feels that something wrong.

Bug or rtfm ?

**Comment by gfduszynski** [comment]
Was someone able to confirm this ?

My general poking around suggests that at the time of handling the emitted event the router responsible for calling changeName does not exist yet. Therefore this explains why calling **route.exec()** helped.
It triggered emit just after mount if I understand correctly, when the router was finally initialized.

I failed to find any kind of info about this in the docs.

Would be possible/acceptable to introduce a flag of sorts for 
the router.create([bool: evaluateCurrentPath])
for the purpose of processing just this router after mounting of it's component ?

**Comment by gfduszynski** [comment]
Let me know if I can help :)

**Comment by GianlucaGuarini** [comment]
can you please reproduce the issue in a jsfiddle or on something we can actually test? Thanks

**Comment by PeteCoward** [comment]
I too ran across this behaviour. Does this illustrate the problem well enough @GianlucaGuarini?
http://embed.plnkr.co/W5VUSaeDgjNHdOTQGDWZ/

In my project calling *route.exec()* in the child tag mount function worked around the behaviour

**Comment by PeteCoward** [comment]
plunker with workaround https://embed.plnkr.co/YWnUIwGrevkk2ebIWfjQ/

Add this in the child tag
```
    <script>
        var tag = this;
        tag.on('mount', function() {
            route.exec();
        });
    </script>
```

Note this only seems to work for one level of nested tag routing

- tag routing works fine
  - nested tag routing needs a route.exec() in the mount function
    - double nested tag routing I wish you the best of luck

**Comment by virtuald** [comment]
Similar problem, but adding the `route.exec()` fixes it. Would be nice if this worked without issues, the tag-based routing is really intuitive to understand.

**Comment by virtuald** [comment]
This is still pretty lightly tested, but I reworked the tag routing stuff so that it works correctly recursively. You can also put multiple child routers on a tag, and they'll all get triggered correctly. It's very much a break from how the current router tags behave, but it seems to work pretty well and is more intuitive to me. 

I don't have much time to clean it up or PR it, but hopefully someone finds this useful:

https://gist.github.com/virtuald/ac2fabcc37aeb00ecfe7a94e29035c72

**Comment by virtuald** [comment]
I've updated my gist, it had some inconsistencies in how routes were being propagated -- sometimes the route event was top to bottom (initially), then sometimes it was bottom to top (after mount). Made a host of changes that make it behave a lot more reliably I think.

**Comment by GianlucaGuarini** [comment]
Fixed in riot-route@3.1.4 many thanks to @aadilhasan
